### PR TITLE
gnupg1compat: fix compatibility with gnupg20

### PR DESCRIPTION
**Pull request by charles-dyfis-net**
###### Motivation for this change

Presently, gnupg1compat only works with gnupg22. Without this change, the error

```
error: attribute 'version' missing, at .../nixpkgs/pkgs/tools/security/gnupg/1compat.nix:4:26
```

is emitted when evaluating

```
pkgs.gnupg1compat.override { gnupg = pkgs.gnupg20; }'
```

Even with that done, `1compat` after e34ce9d1c551fb43742aada6bb43ccb1a52e64a1 no longer actually created `gpg->gpg2` or `gpgv->gpgv2` symlinks at all, making it effectively a noop (and thus ineffective for the `gnupg20` package, which -- unlike `gnupg22` -- was not modified in the aforementioned commit to create `gpg` binaries itself).

The symlink-creation behavior is thus reintroduced, but if-and-only-if actually needed.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by charles-dyfis-net** [comment]
With 7fa8ebd5610b7e0b346c2bbcf2ac4e7c2609b1a3, a `gpg` symlink is created when wrapping gnupg20, such that gnupg1compat is able to actually fulfill its function.

File: pkgs/tools/security/gnupg/1compat.nix [hunk 1]
```diff
@@ -4,14 +4,22 @@ stdenv.mkDerivation {
...
+    rm $out/bin
+    mkdir -p $out/bin
+    ln -s "${gnupg}/bin/"* $out/bin
+
+    # Add symlinks for any executables that end in 2 and lack any non-*2 version
+    for f in $out/bin/*2; do
+      [[ -x $f ]] || continue # ignore failed globs and non-executable files
+      [[ -e ''${f%2} ]] && continue # ignore commands that already have non-*2 versions
+      ln -s -- "''${f##*/}" "''${f%2}"
+    done
```
Comment (lsix) [inline pkgs/tools/security/gnupg/1compat.nix]: Do you expect other binaries to be symlinked other than `gpg` and `gpgv`?

If no, symlinking them explicitly could be a more convenient to read I think, but what you propose is valid too.

Comment (charles-dyfis-net) [reply pkgs/tools/security/gnupg/1compat.nix]: I don't have particular reason to expect others, but the intent is rolling with whatever changes take place in the upstream packages without needing to maintain code that explicitly tests for individual binaries; if a new executable with a `2` suffix shows up in a gnupg package we could potentially be instructed to wrap in the future, with my user hat on, I would expect that to be symlinked as well.

(But then, as a counterargument, the gpg 1.x series is no longer undergoing development, so there shouldn't be new binaries added there, so the set of links potentially needed for *compatibility* is bounded, and could well be hardcoded).

So... I can see two sides here. If as a reviewer you want to direct a rewrite with more explicit code, I'm happy to do that.

Comment (lsix) [reply pkgs/tools/security/gnupg/1compat.nix]: This is good for me as it is. It is just less easily readable for non bash gurus

**Pull request merged**